### ISSUE TITLE
Autodrive does not wait for bunker silo unloading to finish (with quick fix)

### ISSUE THREAD

**Issue by cyr123** [Stephan-S/FS22_AutoDrive#928]
I found that when I have multiple vehicles queued up to unload into a bunker they will not wait for but instead proceed to back into the bunker and block, crash into and/or dump chaff onto the one currently unloading.

They *will* wait while the vehicle in front backs into the bunker, but then start to drive again once the unloading starts.
I had a quick poke around the code and found that this is because the collision check that looks to see if the path ahead is free looks at the current waypoint of the other vehicles, and once a vehicle reaches the bunker and starts to unload the waypoint list is cleared and the vehicle becomes "invisible" to other vehicles.

As a quick fix I commented out two lines in DrivePathModule.lua:

```
function ADDrivePathModule:reachedTarget()
    self.atTarget = true
--    self.wayPoints = nil
--    self.currentWayPoint = 0
end
```

This appears to fix the problem, but I'm sure it will have other consequences.

**Comment by Iwan1803** [comment]
No screenshots of the AD settings and the bunker route...

**Comment by cyr123** [comment]
Apologies, here are screenshots with hopefully all that's needed. The route is just a single reverse-in, drive-out route and I just used two tractors for testing. I've used two-way (blue) segments but I also tried reverse (cyan) segments all the way into the bunker with the same result.

When the unloading tractor is driving (reverse or forward) the second one does stop, just not during the actual unload. This makes sense from my (limited) understanding of the code.

![mpv-shot0011](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/15146695-4ec5-4e57-b276-1a213b3ea5e4)

![mpv-shot0012](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/8bd50979-a4ea-49cf-a134-7e621662bec1)

![mpv-shot0013](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/2b349d00-f75b-4d3b-a600-a9e525210a1d)

![mpv-shot0014](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/87b271ed-8fc4-43c6-bae6-961e11b11f55)

![mpv-shot0015](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/614a744f-8a9c-4c33-a5be-3598f71149a2)

![mpv-shot0016](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/1033e9d8-8bc5-4069-aea1-a829060f43f5)

![mpv-shot0017](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/352b6fd2-3a23-4480-8af2-fa5c175ac16a)

![mpv-shot0018](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/101d21b7-93a0-420c-9738-4186300da90a)

![mpv-shot0019](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/7c2bfbac-5bfa-4c2b-a53e-b0463fba0da2)

![mpv-shot0021](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/8c6588e2-269f-4d07-8e8a-75fec24e202e)

![mpv-shot0023](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/46292960-a94f-42d7-b0db-927469bcb6b3)

![mpv-shot0024](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/84d26f66-a9a6-4e49-99f1-be4b1cf85ecc)

![mpv-shot0025](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/a278240e-3469-4f56-a9a7-5c1f5d3e25bc)

![mpv-shot0026](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/3e6558e4-71ef-4f2f-a9a7-c0af5d3dc1c2)

![mpv-shot0027](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/ea88e322-2b79-4ab9-a7b8-bc2c8ebd2405)

![mpv-shot0029](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/aa9dea5c-baad-48b2-b41f-b99aca6a092a)

![mpv-shot0032](https://github.com/Stephan-S/FS22_AutoDrive/assets/3134704/7b11a47c-fe4b-4be8-acc9-55eda14a914f)

**Comment by cyr123** [comment]
I realize the area is a bit "messy" with other routes making it hard to see but it's pretty straight forward, the incoming route connects to the bidirectional path with a single reverse (cyan) segment at the same node as the outgoing path connects to it.

**Comment by Iwan1803** [comment]
I would suggest 2 Lines and distributed unloading, maybe with smaller trailers.
Removing lines could cause problems in other situations, but @Axel32019 should check this.

**Comment by cyr123** [comment]
I may look into another way to make the unloading vehicle keep the path "reserved" for the purpose of collision checking without affecting anything else.

Thank you to the devs for this awesome mod BTW, it has some "quirks" shall we say but it still allows us to do some awesome stuff in the game that wouldn't be possible otherwise.

**Comment by Axel32019** [comment]
already implemented in 2.0.1.6, but get a further fix #1149 in next release.
